lib/ui/window/pointer_data_packet_converter.cc

```diff
@@ -74,7 +74,42 @@ void PointerDataPacketConverter::ConvertPointerData(
         break;
       }
       case PointerData::Change::kAdd: {
-        FML_DCHECK(states_.find(pointer_data.device) == states_.end());
+        auto iter = states_.find(pointer_data.device);
+        if (iter != states_.end()) {
+          // Synthesizes a remove event if the pointer was not previously
+          // removed.
+          PointerState state = iter->second;
+          PointerData synthesized_data = pointer_data;
+          synthesized_data.physical_x = state.physical_x;
+          synthesized_data.physical_y = state.physical_y;
+          synthesized_data.pan_x = state.pan_x;
+          synthesized_data.pan_y = state.pan_y;
+          synthesized_data.scale = state.scale;
+          synthesized_data.rotation = state.rotation;
+          synthesized_data.buttons = state.buttons;
+
+          if (delegate_.ViewExists(state.view_id)) {
+            synthesized_data.view_id = state.view_id;
+
+            if (state.is_down) {
+              // Synthesizes cancel event if the pointer is down.
+              PointerData synthesized_cancel_event = synthesized_data;
+              synthesized_cancel_event.change = PointerData::Change::kCancel;
+              synthesized_cancel_event.synthesized = 1;
```

Comment:
Should we move this line up to where `synthesized_data` properties are set (line 89) so that we can remove lines 98 and 108? I don't feel strongly about this, feel free to skip

Reply:
Yes, done